### PR TITLE
add prettier, update gitignore

### PR DESCRIPTION
**Pull request by Geczy**
this is necessary so others (like me) can contribute and we won't have conflicting styling and whitespaces etc

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -1,27 +1,35 @@
...
-    // todo: rename events
-    on(event: 'download', listener: (src: string, size: number) => void): this;
-    on(event: 'mkdir', listener: (path: string) => void): this;
-    on(event: 'progress', listener: (src: string, written: number, size: number) => void): this;
-    on(event: 'done', listener: (src: string) => void): this;
-    on(event: 'sshBegin', listener: () => void): this;
-    on(event: 'sshFinish', listener: () => void): this;
-    on(event: 'patch', listener: (remote: string) => void): this;
+    pack(suggested: import("fs").PathLike | null): Promise<string>;
+    #private;
```
Comment (ChiChou) [inline types/index.d.ts:30]: It's not private at all

Comment (Geczy) [reply types/index.d.ts:30]: idk i just ran the typegen script

File: types/lib/installation.d.ts
Comment (ChiChou) [inline types/lib/installation.d.ts:1]: I deleted the auto generated types intentionally because these are internal libraries

Comment (Geczy) [reply types/lib/installation.d.ts:1]: then they should all be gitignored

**Comment by ChiChou** [comment]
And, I am a singlequote man

**Comment by Geczy** [comment]
it's ready now

**Comment by ChiChou** [comment]
Please no autogenerated PRs like this next time, feature contributions are prefered. Appeciated that!

**Pull request closed without merging**